### PR TITLE
Fix the linker libraries search path for windows

### PR DESCRIPTION
**Pull request by vsebe**
Update libraries path for microsoft linker

Issue: #18

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by vsebe** [comment]
@Jenkins test sanity plinux

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -19,8 +19,13 @@
...
 [
-  JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
-  OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+  if test "x$OPENJDK_BUILD_OS_ENV" = xwindows.cygwin; then
+    JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -libpath:\$(SUPPORT_OUTPUTDIR)/../vm/lib"
+    OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -libpath:\$(SUPPORT_OUTPUTDIR)/../vm/lib"
+  else
+    JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+    OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+  fi
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: It might be clearer if we assigned `-L` or `-libpath` to a variable to be used in the flags definitions.

Comment (vsebe) [reply closed/autoconf/custom-hook.m4]: I tried to follow the openjdk style - see flags.m4 
e.g.
```
if test "x$TOOLCHAIN_TYPE" = xmicrosoft; then
    $2JAVA_BASE_LDFLAGS="${$2JAVA_BASE_LDFLAGS} \
        -libpath:${OUTPUT_ROOT}/support/modules_libs/java.base"
    $2JDKLIB_LIBS=""
  else
    $2JAVA_BASE_LDFLAGS="${$2JAVA_BASE_LDFLAGS} \
        -L\$(SUPPORT_OUTPUTDIR)/modules_libs/java.base"
```

Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: Fair enough.

Comment (vsebe) [reply closed/autoconf/custom-hook.m4]: The TOOLCHAIN_TYPE variable is not available for the EARLY_CUSTOM_HOOK (see configure.ac L102). It is set in TOOLCHAIN_DETERMINE_TOOLCHAIN_TYPE (see configure.ac L170), thus the different condition.

Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: Why do the Windows paths have the extra `/lib` segment?
Where is `OPENJDK_BUILD_JAVA_BASE_LDFLAGS` used?

**Comment by vsebe** [comment]
@Jenkins test sanity plinux

File: closed/autoconf/custom-hook.m4 (hunk 1 above)
Comment (vsebe) [reply closed/autoconf/custom-hook.m4]: To set the LDFLAGS_JDKLIB

**Comment by keithc-ca** [comment]
Assuming the sanity test is fine, this looks good to merge.

File: closed/autoconf/custom-hook.m4 (hunk 1 above)
Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: In what file(s)?

**Pull request merged**